UI/ScottPlot/DataPersistedSample/ScottPlotDataPersistedSample/MainPage.xaml.cs

```diff
@@ -0,0 +1,324 @@
+        }
+
+        lock (_seriesLock)
+        {
+            // Clear cache
+            _cachedSeries.Clear();
+        }
+
+        // Clear database
+        await _dbHelper.ClearAllDataAsync();
+
+        // Clear the plot asynchronously
+        await PlotDataAsync();
+        // Update status after clearing data
+        UpdateStatusText(0);
+
+        // Force garbage collection to ensure unused data from previous plots or operations is cleaned up.
+        // This is useful when clearing large datasets here to free up memory.
+        GC.Collect();
+    }
+
+    // Change chart type and refresh plot asynchronously
+    private async void ChangeChartTypeButton_Click(object sender, RoutedEventArgs e)
+    {
+        if (!_isInitialized)
+        {
+            await UpdateUIStatusAsync("Initialization in progress. Please wait.");
+            return;
+        }
+
+        // Cycle through plot types
+        _currentChartTypeIndex = (_currentChartTypeIndex + 1) % _plotTypes.Length;
+        currentPlotType = _plotTypes[_currentChartTypeIndex];
+
+        // Save the selected plot type in the database
+        _dbHelper.SaveLastUsedPlotType(currentPlotType);
+
+        // Plot data asynchronously and update status text with the new chart type
+        await PlotDataAsync();
+        // Update status after changing chart type
+        UpdateStatusText(_cachedSeries.Sum(s => s.DataPoints.Length));
+    }
+
+    // Update the status text with total points and the current chart type
+    private void UpdateStatusText(int totalPoints)
+    {
+        // Marshaling UI updates back to the main thread
+        _dispatcherQueue.TryEnqueue(() =>
+        {
+            StatusTextBlock.Text = $"Total data points: {totalPoints:N0} | Current Chart Type: {currentPlotType}";
+        });
+    }
+
+    // Helper method to update UI status with proper thread dispatching
+    private async Task UpdateUIStatusAsync(string message)
+    {
+        await Task.Run(() =>
+        {
+            _dispatcherQueue.TryEnqueue(() =>
+            {
+                StatusTextBlock.Text = message;
+            });
+        });
+    }
+
+    private Series GenerateRandomWalk(int length, double origin)
+    {
+        double[] data = new double[length];
+        double value = 0;
+
+        for (int i = 0; i < length; i++)
+        {
+            data[i] = value += _random.NextDouble() * 2 - 1;
+        }
+
+        return new Series
+        {
+            DataPoints = data,
+            Origin = origin
+        };
+    }
+
+    // Generate heatmap data from a series' data points
+    private double[,] GenerateHeatmapData(double[] dataPoints)
+    {
+        int size = (int)Math.Sqrt(dataPoints.Length);
+        double[,] heatmap = new double[size, size];
+
+        for (int i = 0; i < size; i++)
+        {
+            for (int j = 0; j < size; j++)
+            {
+                heatmap[i, j] = dataPoints[i * size + j];
+            }
+        }
```

Comment:
```suggestion
        dataPoints.AsSpan().CopyTo(MemoryMarshal.CreateSpan(ref heatmap[0, 0], size * size));
```

Reply:
Thanks for the suggestion! However, I decided to stick with the original nested loop approach instead of using `dataPoints.AsSpan().CopyTo(MemoryMarshal.CreateSpan(ref heatmap[0, 0], size * size))`. The reason is that `heatmap` is a 2D array, and using `MemoryMarshal.CreateSpan` with a 1D array like `dataPoints` could lead to memory layout issues. The `CopyTo` method doesn't handle the mapping between the 1D and 2D array structures, which can cause misaligned data.

Additionally, when I attempted the change, it led to an exception ("Destination is too short"). This happened because the 1D data wasn't correctly mapped into the 2D `heatmap`, and the destination array was not being sized or addressed properly. This is why I kept the original nested loop, to ensures proper placement of the data in the 2D array.